### PR TITLE
activity rest 

### PR DESCRIPTION
**Pull request by octavioamu**
##### Activity endpoint checklist
- [x] pagination with dynamic page size
- [x] serialize foreignkey expansions
- [x] flag/unflag activity
- [x] pin/unpin post
- [x] vote poll
- [x] favorite/unfavorite
- [x] like/unlike
- [x] award comment
- [x] filter activities by foreignkey relations + type
- [x] increment view count
- [x] optimize queries with select_related & prefetch_related
- [x] handle permissions - IsOwnerOrReadOnly, CanPin

##### Comment endpoint checklist
- [x] comment crud
- [x] like comment

##### Comment endpoint checklist
features keep / removal
https://hackmd.io/@drufPD6EQXmFyo_fMCZP6g/SJxUj4J0d
##### Refers/Fixes

[GITC-172](https://gitcoin.atlassian.net/browse/GITC-172)
[GITC-173](https://gitcoin.atlassian.net/browse/GITC-173)

### CODE REVIEW

File: app/dashboard/router.py [hunk 1]
```diff
@@ -26,25 +26,35 @@
...
     Activity, Bounty, BountyFulfillment, BountyInvites, HackathonEvent, HackathonProject, Interest, Profile,
-    ProfileSerializer, SearchHistory, TribeMember, UserDirectory,
+    SearchHistory, TribeMember,
 )
 from .tasks import increment_view_count
 
 logger = logging.getLogger(__name__)
 
 
+class ProfileSerializer(FlexFieldsModelSerializer):
```
Comment (octavioamu) [inline app/dashboard/router.py:44]: notice this name is used already on models

Comment (chibie) [reply app/dashboard/router.py:44]: i removed it from models

**Comment by chibie** [comment]
mini api doc

```
api/v0.1/activities - GET, DELETE (to expand foreignkey fields do -> api/v0.1/activities/?expand=~all)
    * api/v0.1/activities/{id} - GET, DELETE 
    * api/v0.1/activities/{id}/flag - POST, DELETE
    * api/v0.1/activities/{id}/pin - POST - e.g { 'what': 'connect' }
    * api/v0.1/activities/{id}/vote - POST - e.g { 'choice': 43 }
    * api/v0.1/activities/{id}/favorite - POST, DELETE
    * api/v0.1/activities/{id}/like - POST, DELETE
    * api/v0.1/activities/{id}/award - POST
    
api/v0.1/comments/?activity={id} - GET
    * api/v0.1/comments - POST
    * api/v0.1/comments/{id} - GET, PUT, PATCH, DELETE
    * api/v0.1/comments/{id}/like - POST, DELETE

pagination -> frontend can specify a page size
    * api/v0.1/activities/?page_size={size}

filtering -> allow filtering by all foreignkeys
    * api/v0.1/activities?(bounty|grant|hackathonevent|project|profile|kudos|kudos_transfer|tip|subscription)=58
    * for example, we can query a bounty with id=58 by fetching api/v0.1/activities?bounty=58

db query optimizations (number of queries)
* comments endpoint - 8 -> 5 -> ??
* activity endpoint - 108 -> 96 -> 73 -> ??
```

**Comment by thelostone-mc** [comment]
Closing this out @octavioamu  <3

**Pull request closed without merging**